### DISCUSSION TITLE
怎么把cmd或powershell等控制台程序嵌入wpf程序中

### DISCUSSION THREAD

**Discussion by AIKA1024** [BYJRK/DotNet-Discussions#46] [WPF 开发]
我想让我的程序里有控制台让用户直接交互，就像vscode那样，但我不想用process来创建控制台程序后，隐藏窗口重定向输入输出流用一些文本控件去展示信息，因为这样我还要判断哪些文字是可删的、用户输入修改颜色等命令都要我维护，所以我想直接嵌个控制台进去。有办法做到吗

**Comment by BYJRK** [comment]
如果是觉得 Process 不够好用的话，或许可以试试 CliWrap

https://github.com/Tyrrrz/CliWrap

**Comment by Dynesshely** [comment]
你可能需要我做的这个库: https://github.com/Crequency/Common.ExternalConsole
视频演示: 【外置控制台, 一个 KitX 开发过程中的衍生产物】 https://www.bilibili.com/video/BV1WR4y1D7Gw/?share_source=copy_web&vd_source=398f740a5323df096e656c1c01553163

**Reply by Dynesshely** [reply]
使用 tcp 与自己的软件通讯 (原来使用 pipe 的版本被我写废了)
输出颜色什么的功能还没有做, 只是用颜色简单区分了一下客户端内容和服务端内容
目前满足了我的调试需求
你也可以 fork 之后加上你需要的功能
